File: C++/Search2.cpp
#include <iostream>
using namespace std;

int main(void) {
    int s[] = {4, 5, 2, 8, 7, 1, 9, 0, 99999};  //配列の最後にとりあえず 99999 を置いておく
    int const N = 8;  // 99999は除いた大きさ
    int d;

    cout << "Input search number: ";
    cin >> d;
    s[N] = d;         //配列の最後(99999の位置)にdを置く

    int i = 0;
    while (s[i] != d) i++;
    if (i == N) {     //iが配列の最後を指していたら…
        cout << "I can't find: " << d << "\n";
    } else {
        cout << "Found: " << d << " at index " << i << "\n";
    }
    return 0;
}

File: Node/Search2.js
// ====== 共通の入出力機能（変更しない）======
require("./io.js");
// ==========================================

async function main() {
    const s = [4, 5, 2, 8, 7, 1, 9, 0];
    const N = s.length;     //番兵を置く前の大きさ

    const d = parseInt(await input("Input search number: "), 10);
    s[N] = d;   //配列の最後に番兵としてdを追加する（JSの配列は自動で伸びるのでダミーは要らない）

    let i = 0;
    while (s[i] !== d) i++;
    if (i === N) {  //iが番兵を指していたら…
        output("I can't find: " + d + "\n");
    } else {
        output("Found: " + d + " at index " + i + "\n");
    }
}

main().finally(close);
